connect/mirror-client/src/main/java/org/apache/kafka/connect/mirror/MirrorClientConfig.java

```diff
@@ -69,6 +74,17 @@ public ReplicationPolicy replicationPolicy() {
         return getConfiguredInstance(REPLICATION_POLICY_CLASS, ReplicationPolicy.class);
     }
 
+    @SuppressWarnings("unchecked")
+    ForwardingAdmin getForwardingAdmin(Map<String, Object> config) {
+        try {
+            return Utils.newParameterizedInstance(
```

Comment:
Could we use `getConfiguredInstance(FORWARDING_ADMIN_CLASS, ForwardingAdmin.class, config);` here?

Reply:
`getConfiguredInstance` works only with `Configurable` the problem with using `Configurable` for this case in my opinion is that it doesn't force inheriting the superclass initializing behaviour.
In the case, the behaviour we want all extended classes to inherit is the initializing of the `delegate = Admin.create(config)` so I used `Utils.newParameterizedInstance`

Reply:
I might be misunderstanding the problem, but the Java language spec forces a subclass to invoker a superclass initializer as the first statement in a constructor, so isn't that sufficient to guarantee that `delegate` is always initialized?

Reply:
Yes, however, based on my understanding `Configurable` class, 
1. `Configurable` usually has no logic in the constructor and doesn't have any parameters passed to the constructor either. Instead, configurations are passed to `Configurable.configure(Map<String, ?> props)` which `getConfiguredInstance` calls after initialize an instance of `Configurable`.

2. Subclasses of `Configurable` aren't forced to implement `configure` and also aren't forced to call `super.configure` as part of the implementation of `configure`. 
 
I can change it to `Configurable` if you prefer this, but my point is that `Configurable` doesn't force the subclasses to inheart the initialization or configure logic.